There is no valid JCache API library at classpath  - fail in multiple tests [HZ-1980]

_master_ (commit 2a6e9365606b8eb8fab9e4751acfb153fdc3ed6e)

Example:
Failed on redhat-8: https://jenkins.hazelcast.com/job/Hazelcast-master-RedHatJDK8/74/testReport/junit/com.hazelcast.client.splitbrainprotection.cache/ClientCacheSplitBrainProtectionReadTest/isClosed_noSplitBrainProtection_splitBrainProtectionType_READ_/
<details><summary>Stacktrace:</summary>

```
java.lang.IllegalStateException: At server, There is no valid JCache API library at classpath. Please be sure that there is a JCache API library in your classpath and it is newer than `0.x` and `1.0.0-PFD` versions!
	at com.hazelcast.client.impl.clientside.ClientICacheManager.getCacheByFullName(ClientICacheManager.java:60)
	at com.hazelcast.client.impl.clientside.ClientICacheManager.getCache(ClientICacheManager.java:47)
	at com.hazelcast.client.splitbrainprotection.cache.ClientCacheSplitBrainProtectionReadTest.cache(ClientCacheSplitBrainProtectionReadTest.java:59)
	at com.hazelcast.splitbrainprotection.cache.CacheSplitBrainProtectionReadTest.isClosed_noSplitBrainProtection(CacheSplitBrainProtectionReadTest.java:151)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)

```
</details>


<details><summary>Standard output:</summary>

```
Finished Running Test: getLocalCacheStatistics_noSplitBrainProtection[splitBrainProtectionType:READ] in 0.707 seconds.
Started Running Test: isClosed_noSplitBrainProtection[splitBrainProtectionType:READ]
03:06:36,849 ERROR |containsKey_noSplitBrainProtection[splitBrainProtectionType:READ]| - [CacheGetConfigOperation] hz.ClientCacheSplitBrainProtectionReadTest_pedantic_lewin.client.thread-1 - [127.0.0.1]:5704 [lsgcdmnrkf] [5.3.0-SNAPSHOT] Service with name 'hz:impl:cacheService' not found!
com.hazelcast.core.HazelcastException: Service with name 'hz:impl:cacheService' not found!
	at com.hazelcast.spi.impl.NodeEngineImpl.getService(NodeEngineImpl.java:438) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.getService(Operation.java:425) ~[classes/:?]
	at com.hazelcast.splitbrainprotection.impl.SplitBrainProtectionServiceImpl.getSplitBrainProtectionAwareService(SplitBrainProtectionServiceImpl.java:300) ~[classes/:?]
	at com.hazelcast.splitbrainprotection.impl.SplitBrainProtectionServiceImpl.findSplitBrainProtection(SplitBrainProtectionServiceImpl.java:282) ~[classes/:?]
	at com.hazelcast.splitbrainprotection.impl.SplitBrainProtectionServiceImpl.ensureNoSplitBrain(SplitBrainProtectionServiceImpl.java:228) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.ensureNoSplitBrain(OperationRunnerImpl.java:366) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.metWithPreconditions(OperationRunnerImpl.java:228) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:263) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:219) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:618) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:597) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:558) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:241) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:71) ~[classes/:?]
	at com.hazelcast.client.impl.protocol.task.AbstractTargetMessageTask.processInternal(AbstractTargetMessageTask.java:50) ~[classes/:?]
	at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.processMessage(AbstractAsyncMessageTask.java:71) ~[classes/:?]
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:153) ~[classes/:?]
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:116) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_302]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_302]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_302]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
Caused by: com.hazelcast.spi.exception.ServiceNotFoundException: Service with name 'hz:impl:cacheService' not found!
	... 25 more
BuildInfo right after isClosed_noSplitBrainProtection[splitBrainProtectionType:READ](com.hazelcast.client.splitbrainprotection.cache.ClientCacheSplitBrainProtectionReadTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230120', buildNumber=20230120, revision=2a6e936, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'isClosed_noSplitBrainProtection[splitBrainProtectionType:READ](com.hazelcast.client.splitbrainprotection.cache.ClientCacheSplitBrainProtectionReadTest):'
03:06:35, accumulated pauses: 2040 ms, max pause: 687 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-RedHatJDK8/74/

In the mentioned build were 264 failures (almost all for the same reason).

--- a/hazelcast/src/test/java/com/hazelcast/cache/impl/operation/OnJoinCacheOperationTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cache/impl/operation/OnJoinCacheOperationTest.java
@@ -25,12 +25,14 @@ import com.hazelcast.logging.ILogger;
 import com.hazelcast.spi.impl.NodeEngine;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.rules.ExpectedException;
+import org.mockito.MockedStatic;
 import org.mockito.Mockito;
 
 import static org.mockito.ArgumentMatchers.any;
@@ -47,6 +49,8 @@ import static org.mockito.Mockito.when;
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class OnJoinCacheOperationTest {
 
+    private static MockedStatic mockedStatic;
+
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
@@ -56,7 +60,12 @@ public class OnJoinCacheOperationTest {
 
     @BeforeClass
     public static void beforeClass() throws Exception {
-        Mockito.mockStatic(JCacheDetector.class);
+        mockedStatic = Mockito.mockStatic(JCacheDetector.class);
+    }
+
+    @AfterClass
+    public static void cleanupMocks() {
+        mockedStatic.close();
     }
 
     @Before

